tests/unit/lms/services/grouping_test.py

```diff
@@ -75,111 +76,85 @@ def generate_authority_provided_id(self):
+        )
 
-    def test_you_can_have_a_group_and_a_section_with_the_same_id(self, svc):
-        course = factories.Course(application_instance=svc.application_instance)
+        # Load the changes we made in SQLAlchemy
+        db_session.refresh(existing_grouping)
 
-        common_grouping_args = {"lms_id": "same_lms_id", "lms_name": "same_name"}
+        assert groupings == [existing_grouping]
+        assert existing_grouping == Any.object.with_attrs(attrs)
 
-        canvas_group = svc.upsert_with_parent(
-            [common_grouping_args], parent=course, type_=Grouping.Type.CANVAS_GROUP
-        )[0]
-        canvas_section = svc.upsert_with_parent(
-            [common_grouping_args], parent=course, type_=Grouping.Type.CANVAS_SECTION
-        )[0]
-        blackboard_group = svc.upsert_with_parent(
-            [common_grouping_args], parent=course, type_=Grouping.Type.BLACKBOARD_GROUP
-        )[0]
+    @pytest.fixture
+    def parent_course(self, svc, db_session):
+        return factories.Course.create(application_instance=svc.application_instance)
 
-        # We've created three groupings with the same application_instance, parent and lms_id.
-        assert (
-            canvas_group.application_instance
-            == canvas_section.application_instance
-            == blackboard_group.application_instance
-        )
-        assert (
-            canvas_group.parent_id
-            == canvas_section.parent_id
-            == blackboard_group.parent_id
-        )
-        assert canvas_group.lms_id == canvas_section.lms_id == blackboard_group.lms_id
-        # But they have a different type and therefore a different authority_provided_id.
-        assert canvas_group.type != canvas_section.type
-        assert canvas_group.type != blackboard_group.type
-        assert blackboard_group.type != canvas_section.type
-        assert (
-            canvas_group.authority_provided_id != canvas_section.authority_provided_id
-        )
-        assert (
-            canvas_group.authority_provided_id != blackboard_group.authority_provided_id
-        )
-        assert (
-            blackboard_group.authority_provided_id
-            != canvas_section.authority_provided_id
+    @pytest.fixture
+    def existing_grouping(self, parent_course, db_session):
+        lms_id = "existing_id"
+
+        return factories.CanvasGroup.create(
+            lms_id=lms_id,
+            lms_name="existing_name",
+            extra={"existing": "extra"},
+            parent=parent_course,
+            # Construct the authority_provided_id correctly, otherwise we get
+            # a fake one generated by factory boy that doesn't line up
+            authority_provided_id=GroupingService.generate_authority_provided_id(
+                parent_course.application_instance.tool_consumer_instance_guid,
+                lms_id,
+                parent_course,
+                type_=Grouping.Type.CANVAS_GROUP,
+            ),
```

Comment:
I think this awkwardness here is why the previous author used the function to insert this. It's certainly less bother.